src/nwb_conversion_tools/tools/hdmf.py

```diff
@@ -1,7 +1,58 @@
 from typing import Tuple
 
 import numpy as np
-from hdmf.data_utils import GenericDataChunkIterator
+from hdmf.data_utils import GenericDataChunkIterator as HDMFGenericDataChunkIterator
+
+
+class GenericDataChunkIterator(HDMFGenericDataChunkIterator):
+    def _get_default_buffer_shape(self, buffer_gb: float = 1.0) -> Tuple[int]:
+        chunk_bytes = np.prod(self.chunk_shape) * self.dtype.itemsize
```

Comment:
For now, we are locked into using `buffer_gb` and `chunk_mb` since that is what made it into the `hdmf` release. We really prefer to not make backwards incompatible changes to that package unless they are critical since it's the most upstream/general package.

Eventually, it would be nice to fully implement a rigorous way of parsing strings similar, if not identical, to SpikeInterface: https://github.com/SpikeInterface/spikeinterface/blob/ce2af6098ea4e9caf2410bf5f3d8a5708bc56841/spikeinterface/core/job_tools.py#L21-L22

and that additional feature would be a good excuse to deprecate the name.


> I just encountered this code for the first time and my first thought was that it was a pointer to the bytes.

Pointer to the bytes? How do you mean?

Reply:
Yes, keeping it compatible with `hdfm5` seems like a more important concern than code readability. I do think though that we should just re-defined the variables at the beginning for clarity so we used those interanlly for readability. 


You can have a variable that is an actual reference to the [bytes](bytearray).

Reply:
> You can have a variable that is an actual reference to the [bytes](https://github.com/catalystneuro/nwb-conversion-tools/pull/bytearray).

Link is broken - what did you mean to reference?

Reply:
> I do think though that we should just re-defined the variables at the beginning for readability in the rest of the function.

This function isn't intended to live here for any longer than it takes to ensure that it functions well and without error, at which time it will be propagated up to `hdmf`.

Reply:
https://docs.python.org/3/library/functions.html#func-bytes

All right.